controllers/index/xjoinindexvalidator_iteration.go

```diff
@@ -202,45 +202,42 @@ func (i *XJoinIndexValidatorIteration) buildDBConnectionEnvVars(references []src
 			return envVars, errors.Wrap(err, 0)
 		}
 
-		envVars = append(envVars, v1.EnvVar{
-			Name: strings.Split(dataSourcePipelineName, ".")[0] + "_DB_HOSTNAME",
-			ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: dataSourcePipeline.Spec.DatabaseHostname.ValueFrom.SecretKeyRef,
-			},
-		})
+		envVarPrefix := strings.Split(dataSourcePipelineName, ".")[0]
+		hostnameEnvVar, err := dataSourcePipeline.Spec.DatabaseHostname.ConvertToEnvVar(envVarPrefix + "_DB_HOSTNAME")
+		if err != nil {
+			return envVars, errors.Wrap(err, 0)
+		}
+		envVars = append(envVars, hostnameEnvVar)
 
-		envVars = append(envVars, v1.EnvVar{
-			Name: strings.Split(dataSourcePipelineName, ".")[0] + "_DB_USERNAME",
-			ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: dataSourcePipeline.Spec.DatabaseUsername.ValueFrom.SecretKeyRef,
-			},
-		})
+		usernameEnvVar, err := dataSourcePipeline.Spec.DatabaseUsername.ConvertToEnvVar(envVarPrefix + "_DB_USERNAME")
+		if err != nil {
```

Comment:
Observation:  `ConvertToEnvVar` raises an error on empty string, which should not be a case in most (if not all) of the calls. It could raise a `panic` error, as it is a programmatic mistake not an operational error, or not raise anything and let it be raised by k8s (on empty env var name). This way we'd reduce these six err checks.

Reply:
https://issues.redhat.com/browse/ESSNTL-4455